lib/checkstl.cpp

```diff
@@ -1640,27 +1640,44 @@ void CheckStl::missingComparison()
             bool bComparedInAdvance = false;
 
             // Parse loop..
-            for (const Token *tok3 = scope.bodyStart; tok3 != scope.bodyEnd; tok3 = tok3->next()) {
-                if (Token::Match(tok3, "%varid% ++", iteratorId) || 
-                    Token::Match(tok3->previous(), "++ %varid% !!.", iteratorId))
+            for (const Token *tok3 = scope.bodyStart; tok3 != scope.bodyEnd; tok3 = tok3->next()) 
+            {
+                if (tok3->varId() == iteratorId)
                 {
-                    if (!bComparedInAdvance)
-                        incrementToken = tok3;
-                    else
-                        bComparedInAdvance = false;
+                    if (Token::Match(tok3, "%varid% = %name% . insert ( ++| %varid% ++| ,", iteratorId)) 
```

Comment:
For this check I am still using Match() function. I am not sure if this is considered "old style" check and need to be changed. I like flexibility offered by RegEx style pattern matching. That makes code shorter but probably cryptic. On the other hand syntax tree based checks are explicit but verbose. For this check I find "old style" easier. I have a feeling that both types of checks can be used based on which works best for problem at hand. Please correct me if I am wrong. By the way comment on top of SimpleMatch() says it accepts %cmd% pattern. But looking at code it doesn't work. I tried using it and it didn't match.

Reply:
no simpleMatch() does not accept any %cmd% patterns feel free to fix the documentation then later..

Reply:
> I like flexibility offered by RegEx style pattern matching.

Yes. I like that too. We tried to imagine some AST regex before but I do not think any suggestion felt very good. If you have some suggestions feel free to share it..

the problem to use "plain old reg ex" is that it's too explicit. Some extra parentheses and it won't work. An expression instead of %name% as container and it won't work. and so on..

We still use `Token::Match` a lot even when checking the syntax tree.

Reply:
> no simpleMatch() does not accept any %cmd% patterns feel free to fix the documentation then later..

I wasn't reading it carefully. The simpleMatch() help doesn't mention about pattern. There is no need to fix it. Sorry for confusion.